### DISCUSSION TITLE
Does Milvus guarantees exactly once processing for Kafka messages

### DISCUSSION THREAD

**Discussion by divya2661** [milvus-io/milvus#33773] [Q&A and General discussion]
Does Milvus guarantee that any insert or update request will be processed exactly once? If so, does it rely on Kafka's exactly-once semantics, or does it have an internal mechanism to prevent duplicate processing in the event of consumer failure?

**Comment by xiaofan-luan** [comment]
Hi Divya,
I think we have e mechanism to skip rowID consumed before so yes I guess consume is exact once.
